test/cpp/api/dataloader.cpp

```diff
@@ -833,57 +834,83 @@ TEST(DataLoaderTest, RespectsTimeout) {
   ASSERT_LT(duration.count(), 1);
 }
 
+struct Barrier {
```

Comment:
Document me plz :)

Reply:
This doesn't seem to be a barrier in the conventional sense, because different threads blocking on the barrier can get let out earlier than when everyone waits.